keps/sig-release/20190219-publishing-packages.md

```diff
@@ -0,0 +1,317 @@
+    And a release team member ran `./gcbmgr release master --buildversion=${{VERSIONID}} --nomock`
+   When a user inspects the officially documented deb and rpm kubernetes repositories for this specific kubernetes minor version
+   Then they see the newly cut alpha releases published in the alpha channel only
+```
+
+```
+Scenario: End users can get the public key the packages or the repository metadata is signed with
+  Given a user has a system configured with not allowing unsigned untrusted package repositories
+    And they have setup the officially documented repository for a specific kubernetes minor release
+   When they download the public key from the location stated in the official documentation
+    And they configure their system's package manager to use that key
+    And they use their system's package manager to install a package from this specific kubernetes minor release
+   Then their package manager will not complain about untrusted packages, sources or repositories
+```
+
+<!--
+```
+Scenario: [...]
+  Given ...
+    And ...
+   When ...
+   Then ...
+```
+-->
+
+### Implementation Details/Notes/Constraints
+
+Packages will be published for different ...
+- **`${dist}`**: a combination of `<distribution>-<code-name-or-version>`
+  (e.g. `debian-jessie`, `ubuntu-xenial`, `fedora-23`, `centos-7`, ...)
+- **`${k8s_release}`**: the version of kubernetes `<major>.<minor>`
+  (e.g. `1.12`, `1.13`, `1.14`, ...)
+- **`${channel}`**: can be `stable`, `dev`, `nightly`
+    - `stable`: all official releases for `${k8s_release}`
+      (e.g.: `1.13.0`, `1.13.1`, `1.13.2`, ...)
+    - `dev`: all development releases for all minor releases in this `${k8s_release}`, including `alpha`s, `beta`s and `rc`s
+      (e.g.: `1.13.0-rc.2`, `1.13.2-beta.0`, `1.13.1-alpha.3`, ...)
+    - `nightly`: any package cut automatically on a daily basis
+
+
+This means, that end-users can configure their systems’ package managers to use those different `${channel}`s of a kubernetes `${k8s_release}` for their `${dist}`.
+
+A configuration for the package managers might look something like:
+
+- deb:
+    ```
+    # deb http://apt.kubernetes.io/${dist} ${k8s_release} ${channel}
+    deb http://apt.kubernetes.io/debian-jessie 1.13 nightly
+    ```
+- rpm/yum:
+    ```
+    [kubernetes]
+    name=Kubernetes
+    # baseurl=http://yum.kubernetes.io/${dist}/${k8s_release}/${channel}
+    baseurl=http://yum.kubernetes.io/fedora-27/1.13/nightly
+    enabled=1
+    gpgcheck=1
+    repo_gpgcheck=1
+    gpgkey=file:///etc/pki/rpm-gpg/kubernetes.gpg.pub
+    ```
+
+Different architectures will be published into the same repos, it is up to the package managers to pull and install the correct package for the target platform.
+
+
+Ideally, publishing/promoting a package means to commit a change to a configuration file which triggers a tool similar to the [Image Promoter][img-promoter].
+That tool ...
```

Comment:
This seems like a good piece of the implementation. I see 3 options: (1) build the tool from scratch, (2) reuse the image promoter's libraries, or (3) extend the image promoter itself to also take care of non-Docker artifacts (GCS artifacts).

Option (3) feels like the most natural option, but I'd like to hear others' thoughts on this.

Reply:
I think that depends a bit on how/where we end up publishing the debs/rpms. E.g. If we publish directly out of a GCS bucket, things might be a bit different then if we'd publish out of a self-hosted service like aptly or a externally hosted service like packagecloud (those are just examples, I am not saying that we have to do either one of the later).

But in general, I agree: Whatever we end up with, it should be as close to or possibly part of the image promoter.

I will reword the KEP on that. Thanks!